controllers/machine_controller_phases.go

```diff
@@ -110,6 +110,7 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, cluster *clus
 	}
 
 	// Initialize the patch helper.
+	// NOTE: This object doesn't implement observedGeneration. No need to add the WithStatusObservedGeneration option.
```

Comment:
This comment seems a bit misleading, we shouldn't be updating observedGeneration because we are not managing this resource, just trying to get information from an externally reconciled resource.

Reply:
comment changed in: // NOTE: This resource isn't managed by the controller. No need to add the WithStatusObservedGeneration option.